src/Configuration.php

```diff
@@ -134,6 +150,107 @@ public function __construct(array $config = [])
         $this->init();
     }
 
+    /**
+     * Construct a Configuration object from Symfony Console input.
+     *
+     * This is great for adding psysh-compatible command line options to framework- or app-specific
+     * wrappers.
+     *
+     * It is expected that $input has already been bound to an appropriate InputDefinition.
+     *
+     * see self::getInputOptions
+     *
+     * @throws \InvalidArgumentException
+     *
+     * @param InputInterface $input
+     *
+     * @return self
+     */
+    public static function fromInput(InputInterface $input)
+    {
+        $config = new self(['configFile' => $input->getOption('config')]);
+
+        // Handle --color and --no-color (and --ansi and --no-ansi aliases)
+        if ($input->getOption('color') || $input->getOption('ansi')) {
```

Comment:
I'm feeling really good about this pull request. I think this is my last open question:

Should we use `InputInterface::getOption`?

Using `getOption` like we are here means things will blow up if we're handed input that hasn't been bound to a compatible input options definition. It _is_ more correct, though. We just can't really enforce that people give us good input. 

The more pragmatic approach is probably to use the underlying `hasParameterOption` / `getParameterOption` and do some more of the parsing and interpreting work manually, ourselves. It's not as clean, and not as guaranteed-correct, but it also doesn't rely on callers having already bound compatible options. 

Either way, we'd still continue binding the input properly in `bin/psysh` when _we_ call it. We'd just be more forgiving of other callers not doing the same.

I'm leaning towards the second option.

@GrahamCampbell what do you think? This is somewhat relevant to https://github.com/laravel/tinker/pull/98 🙂

Reply:
K. I wen't with a third option.

I've got something that works either way, though definitely better with bound input. See 3c9b112.

It's a bit of a mess, but probably acceptably tucked away from end users that it won't be an issue?